sync-team/src/github/mod.rs

```diff
@@ -732,9 +732,16 @@ impl UpdateRepoDiff {
             return Ok(());
         }
 
-        if self.settings_diff.0 != self.settings_diff.1 {
+        // If we're unarchiving, we have to unarchive first and *then* modify other properties
+        // of the repository. On the other hand, if we're achiving, we need to perform
+        // the archiving *last* (otherwise permissions and branch protections cannot be modified)
+        // anymore. If we're not changing the archival status, the order doesn't really matter.
+        let is_unarchive = self.settings_diff.0.archived && !self.settings_diff.1.archived;
+
+        if is_unarchive {
```

Comment:
```suggestion
        if is_unarchive && self.settings_diff.0 != self.settings_diff.1 {
```
?

Reply:
When `is_unarchive` is true, then the second condition must hold :)